tests/unit_tests/test_frontend.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import ast
+import inspect
+import textwrap
+
+from gt_frontend import ast_node_matcher as anm
+
+
+class TestAstNodeMatcher:
```

Comment:
I feel this test suite could be simplified while still covering more cases by simplifying the `match` function.

Also: is there a reason for this to be a class and not just top-level test functions? There is no setup or teardown it seems. Or is this run using something other than `pytest`?